### DISCUSSION TITLE
Return kube-proxy under Talos control

### DISCUSSION THREAD

**Discussion by den-is** [siderolabs/talos#9631] [Q&A] [answered]
Hello,
Initially, I had to manage kube-proxy manually and deploy two distinct kube-proxy daemonsets in a cluster. One for Linux nodes, the other one for Windows nodes

Now project decided to get rid of windows nodes. And we want to simplify talos cluster deployment and return kube-proxy control to Talos by setting:
```hcl
      proxy = {
        disabled = false
        extraArgs = {
          metrics-bind-address = "0.0.0.0:10249"
        }
      }
```

But I'm really confused and don't know how to proceed with migration procedure.
How to disable existing `kube-proxy-linux` DaemonSet, or maybe it is possible to temporary run two kube-proxies in parallel?

**Comment by frezbo** [comment, marked as answer]
you can just delete custom kube-proxy manifests from api server and run a `talosctl upgrade-k8s` (upgrading to same k8s version, can do `--dry-run` to see the changes) after re-enabling kube-proxy in machine config.

**Reply by den-is** [reply]
I'm afraid of completely loosing network connectivity to the cluster.
But communication and configuration via `talosctl` makes sense.
Also I have talos deployed using terraform. I remember there were some issues with `upgrade-k8s` operation via terraform

**Reply by den-is** [reply]
It worked fine even withing terraform setup.
Marking your answer, since it was the first

**Comment by smira** [comment]
I don't think you can run two kube-proxy daemons on the same machine.

But also stopping kube-proxy won't kill existing connections, it will stop reconciliation (creating/removing iptables rules).

So probably I would nuke your own daemonset, and re-enable kube-proxy on Talos side, Talos will immediately re-create the daemonset.
